### ISSUE TITLE
client_secret and clientSecret

### ISSUE THREAD

**Issue by jeffwhelpley** [hapijs/bell#102]
So, it looks like clientSecret was changed to client_secret here:

https://github.com/hapijs/bell/commit/96ef98c422dd8d15964319bac9b8189d303278dd#diff-84fad1d3b3913ec86805e857580c5d54R199

and that breaks my code since when I try to set client_secret I get:

```
child "clientSecret" fails because ["clientSecret" is required] {"name":"ValidationError","details":[{"message":"\"clientSecret\" is required","path":"clientSecret","type":"any.required","context":{"key":"clientSecret"}}],"_object":{"ttl":31536000000,"domain":".gethuman.com","isSecure":false,"password":"MIICXgIBAAKBgQC4xBeP3h36ikta","cookie":"bell-facebook","clientId":"523625311006913","client_secret":"e8f655bd2b7ebecf977c274cd0c915dc","provider":{"useParamsAuth":true,"protocol":"oauth2","auth":"https://www.facebook.com/v2.3/dialog/oauth","token":"https://graph.facebook.com/v2.3/oauth/access_token","scope":["email"]},"name":"custom"}}
```

but then if I set clientSecret as I have been doing, I get:

```
Error validating client secret.
```

In debugging my Bell I can see that this is because client_secret is null here:

https://github.com/hapijs/bell/blob/master/lib/oauth.js#L199

I don't mind submitting a pull request to fix this but I noticed that clientSecret is actually used in other places of the code so perhaps there is something I am missing. If it is OK with you I would very simply submit a pull request to do:

```
query.client_secret = settings.client_secret || settings.clientSecret;
```

**Comment by qualquervalor** [comment]
I think it might have been a typo
https://github.com/hapijs/bell/pull/101

**Comment by jeffwhelpley** [comment]
Yeah, I figured as much. Thanks @qualquervalor for submitting a PR!

**Comment by ldesplat** [comment]
Sorry about that. I had tested the code with my own provider but I was using Authorization header which used the right variable :(

**Comment by jeffwhelpley** [comment]
@ldesplat no worries. One thing. Unless I am mistaken it looks like the PR from @qualquervalor was merged, but the version number hasn't been incremented and published to npm.

**Comment by chefArchitect** [comment]
Hi guys! Are you planning a release with this fix soon?  I've just discovered the issue and couldn't login with most of the networks due to this error.

**Comment by Icehunter** [comment]
@hueniverse I was wondering if the NPM/package version can be incremented from the change here: https://github.com/hapijs/bell/blob/master/lib/oauth.js#L199 ?

4.0.0 does not have that line and is currently breaking on facebook and other providers.

Thanks for all the work! :)

**Comment by hueniverse** [comment]
@geek Care to publish a new version with the fix?

**Comment by ericeslinger** [comment]
Yep, I'd sure love to see a new version, I independently found this bug and was in the midst of forking to submit a PR when I found it had already been fixed in master. I rolled back to 3.0.0 in the meantime.

**Comment by maxbeatty** [comment]
Sorry to pile on but like others I discovered this during development, found my way here, and would appreciate publishing a new version with the fix. Thanks! :ramen:

**Comment by alanshaw** [comment]
+1 please publish the fix

**Comment by thebillkidy** [comment]
+1, fix is needed, facebook auth doesn't work without it

**Comment by ldesplat** [comment]
Fixed by #101 and #107
